### ISSUE TITLE
Cannot authenticate with magento repo using composer?

### ISSUE THREAD

**Issue by ipkpjersi** [zepgram/magento2-fast-vm#73]
Hello,

I am using Ubuntu 18.04 with PHP 7.4 and trying to use this repo to install Magento 2.4.

I seem to get an error when trying to authenticate with the magento repo using composer:
![](https://i.imgur.com/pZRHSGD.png)

However, I have entered my magento credentials in config.yaml and the magento credentials I am using work for accessing www.magentocommerce.com/products/downloads/info/help since it does not give me a 401 Unauthorized error:
![](https://i.imgur.com/cyFX8NA.png)

Any suggestions?

Here is my config.yaml (I also had problems with it getting stuck on NFS mount step so I am trying rsync instead):
```yaml
vmconf:
  machine_name: 'Vagrant Magento 2'
  network_ip: '192.168.200.50'
  host_name: 'zepgram'
  memory: '4096'
  cpus: '1'
  mount: 'rsync'
  path: 'root'
  provision: 'all'
composer:
  username: 'MAG#HERE'
  password: 'CORRECTKEYHERE'
git:
  name: 'John Doe'
  email: 'john@doe.com'
  host: 'github.com'
  repository: 'ssh://git@github.com:project-name.git'
magento:
  url: 'dev.magento.com'
  source: 'composer'
  edition: 'community'
  version: '2.4.0'
  php_version: 'default'
  sample: 'true'
  mode: 'developer'
  currency: 'USD'
  language: 'en_US'
  time_zone: 'America/New_York'
  crypt_key: ''
```

**Comment by zepgram** [comment]
Hi,

I got the same issue, I'm investigating.

**Comment by zepgram** [comment]
Ok I made some research and this is related to the recent upgrade from composer v1.x to [composer v2.x](https://blog.packagist.com/composer-2-0-is-now-available/) released the 24 october.
For now, it looks like Magento is not compatible with this new composer version.

I don't know who is responsible of this, it could be on Magento side or maybe Composer. Until they fix this, I will lock composer to version 1.x to keep this vagrant stable.

PS: For NFS you should be able to make it work by installing  `apt install nfs-kernel-server` and running your credentials at first setup.

**Comment by ipkpjersi** [comment]
I did have nfs-kernel-server installed, I'm not sure how to run my credentials at first setup.

**Comment by zepgram** [comment]
You should have this on first vagrant up:
![image](https://user-images.githubusercontent.com/16258478/97120454-5bfe2100-1717-11eb-8598-d03099503cae.png)

**Comment by zepgram** [comment]
I just read your issue and there is no "MAGEID" required to authenticate with composer.

You can generate/get them from this url: 
https://marketplace.magento.com/customer/accessKeys/

**Comment by ipkpjersi** [comment]
I did have it prompt me for permissions for /etc/exports and it was created successfully but it timed out connecting to NFS so I switched it to rsync.

I did authenticate the access keys and I was able to authenticate with http://www.magentocommerce.com/products/downloads/info/help on my host but I was not able to authenticate through this Vagrant box, thus magento did not install for me. However, after you downgraded it to composer version 1, it is now working for me.

I did notice this when doing `vagrant up`, I'm not sure if it matters, I am guessing it doesn't:
![](https://i.imgur.com/wNdpjyA.png)

Since this is now working for me, I will now close this issue.
